### PR TITLE
CPLAT-15997 Scope Queries to the Body Element

### PR DESCRIPTION
**Pull request by joebingham-wk**
## Motivation
  

I ran into an issue where I needed to query a portal that was attached directly to the `document.body` rather than inside of RTL's `container`. Doing research revealed that [this should work out of the box](https://github.com/testing-library/react-testing-library/issues/62#issuecomment-506005319) because queries are bound to [the document.body by default](https://github.com/testing-library/react-testing-library/blob/8f17a2bc4fa9c2c17a623b38ded9ab18b086e457/src/pure.js#L98). We were doing things slightly differently by binding queries to the render result's container, preventing us from being able to access portals like the library anticipates. This PR tweaks that to align with the JS side!

## Changes
  
- Change the default query bindings to be the render result's `baseElement` instead of `container`
- Update tests

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author:
        
      - [ ] The test changes make sense and are expected
      - [ ] CI passes
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/react_testing_library/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/react_testing_library/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: test/unit/dom/matchers/has_form_values_test.dart [hunk 1]
```diff
@@ -39,7 +39,8 @@ void main() {
   group('hasFormValues matcher', () {
     const rootElemTestId = 'root-of-test-form';
     RenderResult renderFormWithValues(
-        /*ReactDomComponentFactoryProxy*/ dynamic rootElem, List<_FormElemDefinition> els) {
+        /*ReactDomComponentFactoryProxy*/ dynamic rootElem,
+        List<_FormElemDefinition> els) {
```
Comment (joebingham-wk) [inline test/unit/dom/matchers/has_form_values_test.dart:43]: These are formatting changes.. I'm not sure why they're there (I formatted on 2.13.4). The only real change is in `render.dart` and then any test files

**Review by aaronlademann-wf — APPROVED** [review]
+1 / Semver +1

**Comment by greglittlefield-wf** [comment]
QA +1

**Review by kealjones-wk — APPROVED** [review]
+1

**Comment by aaronlademann-wf** [comment]
@Workiva/release-management-pp

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**